akka-remote/src/main/scala/akka/remote/artery/RemotingFlightRecorder.scala

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2018-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.remote.artery
+
+import java.net.InetSocketAddress
+
+import akka.actor.Address
+import akka.actor.ExtendedActorSystem
+import akka.actor.Extension
+import akka.actor.ExtensionId
+import akka.actor.ExtensionIdProvider
+import akka.annotation.InternalApi
+import akka.remote.UniqueAddress
+import akka.util.JavaVersion
+
+import scala.util.Failure
+import scala.util.Success
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+object RemotingFlightRecorder extends ExtensionId[RemotingFlightRecorder] with ExtensionIdProvider {
+
+  override def createExtension(system: ExtendedActorSystem): RemotingFlightRecorder =
+    if (JavaVersion.majorVersion >= 11) {
```

Comment:
Maybe we should also have a config to be able to turn it off completely (NoOpRemotingFlightRecorder). Then the config should be in akka-actor so we prepare for being able to emit events from there also.

Reply:
Added!